### PR TITLE
Refactor: separate bluepy and sonata access objects into different modules

### PR DESCRIPTION
**Pull request by anilbey**
### Changes

1. It separates the implementations of bluepy and sonata access objects into different modules.
2. exclude 2 bluepy modules from coverage. Their tests are separately done on the gitlab CI.

#### Changes in terms of files

* circuit_access.py became 3 files
  - definition.py
  - bluepy_circuit_access.py
  - sonata_circuit_access.py

* simulation_config.py became 3 files
  - definition.py
  - bluepy_simulation_config.py
  - sonata_simulation_config.py

### CODE REVIEW

File: bluecellulab/circuit/circuit_access/bluepy_circuit_access.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
+from __future__ import annotations
```
Comment (ilkilic) [inline bluecellulab/circuit/circuit_access/bluepy_circuit_access.py:1]: Shouldn't there be a copyright header present?

Comment (anilbey) [reply bluecellulab/circuit/circuit_access/bluepy_circuit_access.py:1]: Oh thanks let me add.

Comment (anilbey) [reply bluecellulab/circuit/circuit_access/bluepy_circuit_access.py:14]: Done in the last push.

**Review by ilkilic — APPROVED** [review]
Thanks! everything looks good to me

**Pull request merged**